index.js

```diff
@@ -498,6 +498,54 @@ class ServerlessAppsyncPlugin {
         ];
         break;
 
+      case 'RELATIONAL_DATABASE':
+        // Allow any action on the Datasource's table
+        defaultStatement.Action = [
+          "rds-data:DeleteItems",
+          "rds-data:ExecuteSql",
+          "rds-data:GetItems",
+          "rds-data:InsertItems",
+          "rds-data:UpdateItems",
+        ];
+
+        const dbResourceArn = {
+          "Fn::Join" : [
+            ":",
+            [
+              'arn',
+              'aws',
+              'rds',
+              ds.config.region || config.region,
+              { "Ref" : "AWS::AccountId" },
+              'cluster',
+              ds.config.dbClusterIdentifier,
+            ],
+          ],
+        };
+
+        const secretResourceArn = {
+          "Fn::Join" : [
+            ":",
+            [
+              'arn',
+              'aws',
+              'secretsmanager',
+              ds.config.region || config.region,
+              { "Ref" : "AWS::AccountId" },
+              'secret',
+              ds.config.awsSecretStoreArn,
+            ],
+          ],
+        };
+
+        defaultStatement.Resource = [
+          dbResourceArn,
+          secretResourceArn,
+          { "Fn::Join" : [ ":", [dbResourceArn, '*'] ] },
+          { "Fn::Join" : [ ":", [secretResourceArn, '*'] ] },
+        ];
```

Comment:
@roznalex I think something is wrong here.
Look at the doc example: https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-rds-resolvers.html


````json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "rds-data:DeleteItems",
                "rds-data:ExecuteSql",
                "rds-data:GetItems",
                "rds-data:InsertItems",
                "rds-data:UpdateItems"
            ],
            "Resource": [
                "arn:aws:rds:us-east-1:123456789012:cluster:mydbcluster",
                "arn:aws:rds:us-east-1:123456789012:cluster:mydbcluster:*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "secretsmanager:GetSecretValue"
            ],
            "Resource": [
                "arn:aws:secretsmanager:us-east-1:123456789012:secret:mysecret",
                "arn:aws:secretsmanager:us-east-1:123456789012:secret:mysecret:*"
            ]
        }
    ]
}
````

We need 2 statements in the policy, one for accessing the RDS cluster, and one for accessing the secret manager.
But here it looks like you are generating one single statement that can access both resources.
On top of that, I am missing the `secretsmanager:GetSecretValue` action for the secret `SecretsManager`.

I think we need some small refactoring of `getDefaultDataSourcePolicyStatements` here.
It should return an array of statements, instead of one single statement.
For all the pre-existing data sources, nothing changes and it returns an array of one item.
For RDS, it will return 2 statements, one for the RDS endpoint, and another for SecretsManager.

Reply:
Hi, @bboure. Just updated the PR. Could you check it, please?